### ISSUE TITLE
Ported monodepth instability (pytorch)

### ISSUE THREAD

**Issue by alwynmathew** [mrharicot/monodepth#168]
I have ported monodepth to pytorch [here](https://github.com/alwynmathew/monodepth-pytorch).

But when I train the quality of disparity degrades as I train more. Epoch 1 - 5 give promising results but it worst as I goes on. I have also open an [issue](https://github.com/alwynmathew/monodepth-pytorch/issues/1) in my own repo.

Recently I got hands on [another ported monodepth pytorch code](https://github.com/ClubAI/MonoDepth-PyTorch) but I faced the same problem that I did with my code. I have posted an [issue](https://github.com/ClubAI/MonoDepth-PyTorch/issues/1) in that repo too. 

Couldn't figure out why pytorch ported codes are not as good as the original implementation, even though the hyperparameters are the same.

**Comment by alwynmathew** [comment]
learning_rate you used is 0.01 or 0.001(as mentioned [here](https://github.com/ClubAI/MonoDepth-PyTorch/issues/1#issuecomment-399382571))?

**Comment by NikolasEnt** [comment]
@alwynmathew learning rate was 0.01=1e-2 as mentioned in our [README](https://github.com/ClubAI/MonoDepth-PyTorch#train-results) and issue you linked.

**Comment by mrharicot** [comment]
@NikolasEnt Great job with the reimplementation! 
I would recommend using `grid_sample` instead of the handmade bilinear sampler:
https://pytorch.org/docs/stable/nn.html?highlight=grid_sample#torch.nn.functional.grid_sample

**Comment by alwynmathew** [comment]
@NikolasEnt you could use [this](https://github.com/alwynmathew/monodepth-pytorch/blob/master/bilinear_sampler.py) as a starting point for `grid_sample`.

**Comment by NikolasEnt** [comment]
@mrharicot Thank you!
We will consider this function.
